### PR TITLE
Addresser cleanups

### PR DESCRIPTION
**Pull request by karlosz**

### CODE REVIEW

**Review by stylewarning — APPROVED** [review]
looks OK to me

File: src/utilities.lisp [hunk 1]
```diff
@@ -264,3 +264,49 @@ contains the bits of INTEGER. See http://www.cliki.net/ROTATE-BYTE"
   (calculate-instructions-fidelity
    (coerce (parsed-program-executable-code program) 'list)
    chip))
+
+;;;; Queues, taken with permission from https://bitbucket.org/tarballs_are_good/lisp-random/raw/a595eb662c2dce87f362dd0dd2541a93efe9c902/stack-queue.lisp
```
Comment (kilimanjaro) [inline src/utilities.lisp:268]: Can we put this stuff in its own `queue.lisp` or `stack-queue.lisp`? While we can in principle put anything in `utilities.lisp`, I think it's nice for something like a fully-fledged ADT to not get stashed in here with the rest of the rabble.

**Review by kilimanjaro — CHANGES_REQUESTED** [review]
This looks good to me, aside from one minor suggestion/request re: where we stash the queue implementation.

File: src/utilities.lisp (hunk 1 above)
Comment (stylewarning) [reply src/utilities.lisp:268]: I'll invite this as a separate PR. I agree with it.

**Comment by stylewarning** [comment]
(sorry, I merged this a bit too quickly, since Erik depends on these changes)

**Pull request merged**